### PR TITLE
Improve docs on some char boolean methods

### PR DESCRIPTION
**Pull request by ghost**
simple revival of #61794
(also rustfmt on rest of file :)

Documentation for `is_xid_start()` and `is_xid_continue()` couldn't be improved since both methods got remove from this repository

r? @dtolnay 
cc @JohnCSimon

### CODE REVIEW

**Comment by JohnCSimon** [comment]
Ping from triage
@dtolnay can you please review this PR?
cc: @varkor @SimonSapin 

thanks

File: src/libcore/char/methods.rs [hunk 1]
```diff
@@ -392,7 +395,11 @@ impl char {
     #[inline]
     pub fn len_utf16(self) -> usize {
         let ch = self as u32;
-        if (ch & 0xFFFF) == ch { 1 } else { 2 }
```
Comment (varkor) [inline src/libcore/char/methods.rs]: This one was probably okay as-is, but either way is fine.

File: src/libcore/char/methods.rs [hunk 2]
```diff
@@ -436,30 +443,31 @@ impl char {
...
                 *dst.get_unchecked_mut(3) = (code & 0x3F) as u8 | TAG_CONT;
                 4
             } else {
-                panic!("encode_utf8: need {} bytes to encode U+{:X}, but the buffer has {}",
+                panic!(
+                    "encode_utf8: need {} bytes to encode U+{:X}, but the buffer has {}",
                     from_u32_unchecked(code).len_utf8(),
                     code,
-                    dst.len())
+                    dst.len()
```
Comment (varkor) [inline src/libcore/char/methods.rs]:
```suggestion
                    dst.len(),
```

File: src/libcore/char/methods.rs [hunk 3]
```diff
@@ -515,15 +523,24 @@ impl char {
                 *dst.get_unchecked_mut(1) = 0xDC00 | ((code as u16) & 0x3FF);
                 slice::from_raw_parts_mut(dst.as_mut_ptr(), 2)
             } else {
-                panic!("encode_utf16: need {} units to encode U+{:X}, but the buffer has {}",
+                panic!(
+                    "encode_utf16: need {} units to encode U+{:X}, but the buffer has {}",
                     from_u32_unchecked(code).len_utf16(),
                     code,
-                    dst.len())
+                    dst.len()
```
Comment (varkor) [inline src/libcore/char/methods.rs]:
```suggestion
                    dst.len(),
```

**Review by varkor — COMMENTED** [review]
Looks good to me. Just a couple of missing commas.

**Comment by varkor** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**